smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -200,254 +140,142 @@ def __init__(
         """Initialize the WorkerManager
         :param config_loader: Environment config loader that loads the task queue and
         feature store
-        :param workers: A worker to manage
+        :param worker_type: The type of worker to manage
+        :param task_queue: Queue from witch the batched requests have to be pulled
         :param as_service: Specifies run-once or run-until-complete behavior of service
         :param cooldown: Number of seconds to wait before shutting down after
         shutdown criteria are met
         :param comm_channel_type: The type of communication channel used for callbacks
+        :param device: The device on which the Worker should run. Every worker manager
+        is assigned one single GPU (if available), thus the device should have no index.
         """
         super().__init__(as_service, cooldown)
 
-        self._task_queue: t.Optional[CommChannelBase] = config_loader.get_queue()
-        """the queue the manager monitors for new tasks"""
+        self._task_queue = task_queue
```

Comment:
The idea was that - in theory - this could have been any queue (and the dispatcher could or could not be interposed). Given the current state, I think your idea makes more sense.